yt_dlp/extractor/godresource.py

```diff
@@ -0,0 +1,76 @@
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    determine_ext,
+    str_or_none,
+    unified_timestamp,
+    url_or_none
+)
+from ..utils.traversal import traverse_obj
+
+
+class GodResourceIE(InfoExtractor):
+    _VALID_URL = r'https?://new\.godresource\.com/video/(?P<id>\w+)'
+    _TESTS = [{
+        # hls stream
+        'url': 'https://new.godresource.com/video/A01mTKjyf6w',
+        'info_dict': {
+            'id': 'A01mTKjyf6w',
+            'ext': 'mp4',
+            'view_count': int,
+            'timestamp': 1710978666,
+            'channel_id': '5',
+            'thumbnail': 'https://cdn-02.godresource.com/e42968ac-9e8b-4231-ab86-f4f9d775841f/thumbnail.jpg',
+            'channel': 'Stedfast Baptist Church',
+            'upload_date': '20240320',
+            'title': 'GodResource video #A01mTKjyf6w',
+        }
+    }, {
+        # mp4 link
+        'url': 'https://new.godresource.com/video/01DXmBbQv_X',
+        'info_dict': {
```

Comment:
An md5 check can be added for this (they don't work for hls)
```diff
        'url': 'https://new.godresource.com/video/01DXmBbQv_X',
+       'md5': 'xxx',
```

Reply:
Fixed in 4741aded6fda773eefc3b6d53335ea26b9d36b4a